feat: top-level keystore fields

spypsy opened the pull request:
implementation for [A-290](https://linear.app/aztec-labs/issue/A-290/consider-top-level-publisher-for-aztec-node-keystore) proposal

Fixes A-290

PhilWindle commented on yarn-project/node-keystore/src/loader.ts:
```diff
@@ -230,8 +236,31 @@ export function mergeKeystores(keystores: KeyStore[]): KeyStore {
+
+        // When merging v1 validators into a v2+ result, preserve original fallback behavior
+        // by explicitly setting publisher/coinbase/feeRecipient if they're missing
+        if (keystore.schemaVersion === 1 && schemaVersion === 2) {
```
This feels like it is going to get ugly very easily. Why do we need to do this?

I think if somebody tries to merge 2 keystores with different versions we should just throw.

spypsy replied:
yep makes sense. I was on the fence about it as well. Updating

spypsy committed:
feat: top-level keystore fields

implementation for [A-290](https://linear.app/aztec-labs/issue/A-290/consider-top-level-publisher-for-aztec-node-keystore) proposal
